assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,238 @@
+			selected: false,
+			...optionalKeys,
+		};
+	} );
+
+	const initialState = keyBy( mappedChoices, 'answer_ordinal' );
+
+	const [ selectedValues, setSelectedValues ] = useState( initialState );
+
+	// eslint-disable-next-line camelcase
+	const handleCheck = ( answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				selected: ! selectedValues[ answer_ordinal ].selected,
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	// eslint-disable-next-line camelcase
+	const handleAnswerChange = ( event, answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				answer_text: event.target.value?.slice(
+					0,
+					MAXIMUM_CHARACTER_LIMIT
+				),
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleSubmit = () => {
+		const answer = Object.values( selectedValues )
+			.filter( ( { selected } ) => selected )
+			// eslint-disable-next-line camelcase
+			.map( ( { answer_ordinal, answer_text } ) => {
+				// eslint-disable-next-line camelcase
+				if ( answer_text ) {
+					return { answer_ordinal, answer_text };
+				}
+				return { answer_ordinal };
+			} );
+		answerQuestion( {
+			answer,
+		} );
+	};
+
+	const hasEmptySelectedTextValue =
+		// eslint-disable-next-line camelcase
+		choices.filter( ( { write_in, answer_ordinal } ) => {
+			// eslint-disable-next-line camelcase
+			if ( write_in ) {
+				// eslint-disable-next-line camelcase
+				const { selected, answer_text } = selectedValues[
+					answer_ordinal
+				];
+				if ( selected && answer_text.length === 0 ) {
+					return true;
+				}
+			}
+			return false;
+		} ).length > 0;
+
+	const totalSelectedValues = Object.values( selectedValues ).filter(
+		( { selected } ) => selected
+	).length;
+	const hasLessThanMinChoices = totalSelectedValues < minChoices;
+	const hasMoreThanMaxChoices =
+		maxChoices && totalSelectedValues > maxChoices;
+
+	const isSubmitButtonDisabled =
+		hasEmptySelectedTextValue ||
+		hasLessThanMinChoices ||
+		hasMoreThanMaxChoices;
+
+	return (
+		<div className="googlesitekit-survey__multi-select">
+			<SurveyHeader title={ question } dismissSurvey={ dismissSurvey } />
+
+			<div className="googlesitekit-survey__body">
+				{
+					// eslint-disable-next-line camelcase
+					choices.map( ( { answer_ordinal, text, write_in } ) => {
+						const answer = selectedValues[ answer_ordinal ];
+						// eslint-disable-next-line camelcase
+						const uniqueID = `googlesitekit-survey__multi-select-${ answer_ordinal }-${ text }`;
+
+						return (
+							<div
+								key={ text }
+								className="googlesitekit-survey__multi-select__choice"
+							>
+								<Checkbox
+									checked={ answer.selected }
+									onChange={ () =>
+										handleCheck( answer_ordinal )
+									}
+									// eslint-disable-next-line camelcase
+									value={ `${ answer_ordinal }` }
```

Comment:
I'm guessing this prop also needs to be a string? If it's our component we could fix that, but if not let's at least leave a comment saying why we do it.

Reply:
Exactly that!  I'm happy to change `Checkbox` but changing core components is something I wouldn't do lightly :smile: